### PR TITLE
Updated app for enonicXp v7.12.x anf react4xp v4.1.0

### PR DESCRIPTION
**Pull request by annesiri**
Updated npm packages to make feature toggle app work with react4xp v4.1.0 and enonicXp v7.12.0 (to make it work in SSB's enonic environment). 

Notable changes: 
* updated react4xp
* updated to React 18
* changed material ui packages to be compatible with react 18
* updated webpack

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,70 +1,80 @@
...
 }
 
 app {
     name = "${appName}"
     displayName = "${appDisplayName}"
     vendorName = "${vendorName}"
     vendorUrl = "${vendorUrl}"
-    systemVersion = "${xpVersion}"
-}
+    systemVersion = "${xpVersion}"}
```
Comment (omsaggau) [inline build.gradle]: 📐

File: package.json [hunk 2]
```diff
@@ -1,68 +1,59 @@
...
+    "build:client": "webpack --config webpack.client.config.js --color",
+    "build:server": "webpack --config webpack.server.config.js --color",
     "build": "run-p -c build:*",
-    "dist:client": "cross-env NODE_ENV=production webpack --config webpack.client.config.js --color --hide-modules",
-    "dist:server": "cross-env NODE_ENV=production webpack --config webpack.server.config.js --color --hide-modules",
-    "dist": "run-p -c dist:*"
+    "dist:client": "cross-env NODE_ENV=production webpack --config webpack.client.config.js --color",
+    "dist:server": "cross-env NODE_ENV=production webpack --config webpack.server.config.js --color",
+    "dist": "run-p -c dist:*",
+    "build:react4xp": "npm explore @enonic/react4xp R4X_APP_NAME=com.gravitondigital.featuretoggle R4X_DIR_PATH_ABSOLUTE_PROJECT=$INIT_CWD -- npm run build:react4xp"
```
Comment (omsaggau) [inline package.json:14]: Denne kan endres til å bare være `npm explore @enonic/react4xp -- npm run build:react4xp `

File: package.json [hunk 3]
```diff
@@ -1,68 +1,59 @@
...
+    "babel-loader": "^9.1.3",
+    "browserslist": "^4.21.10",
+    "browserslist-config-enonic": "^1.0.8",
+    "cross-env": "^7.0.3",
+    "css-loader": "^6.8.1",
+    "cssnano": "^6.0.1",
+    "file-loader": "^6.2.0",
+    "glob": "^10.3.3",
+    "mini-css-extract-plugin": "^2.7.6",
+    "node-sass": "^9.0.0",
```
Comment (tajakobsen) [inline package.json]: `node-sass` has been deprecated, and I'm not able to build this project on Linux. Can you replace it with the `sass` package?

```suggestion
    "sass": "^1.67.0",
```

**Review by tajakobsen — CHANGES_REQUESTED** [review]
Very good job! 

The only problem is that I'm having problems building it, because I can't find Linux bindings for the deprecated *node-sass*-package. Can you replace it with the *sass*-package. *sass-loader* will work with either.

**Review by tajakobsen — APPROVED** [review]
Looks good. Merging to main.

**Pull request merged**